sphinx/ext/autodoc.py

```diff
@@ -1425,14 +1425,11 @@ def format_args(self):
                 not(inspect.ismethod(initmeth) or inspect.isfunction(initmeth)):
             return None
         try:
-            argspec = getargspec(initmeth)
+            return Signature(initmeth, method=True).format_args()
         except TypeError:
             # still not possible: happens e.g. for old-style classes
             # with __init__ in C
             return None
-        if argspec[0] and argspec[0][0] in ('cls', 'self'):
```

Comment:
autodoc drops first argument if it is named as "cls" or "self".
Is this intended?

Example:
```
class Foo(object):
    """Foo class!"""

    def bar(self, *args, **kwargs):
        """Foo.bar method!"""

    def baz(this, *args, **kwargs):
        """Foo.baz method!"""
```
The source code above is rendered as following:
![2017-06-17 15 30 56](https://user-images.githubusercontent.com/748828/27250766-fe01460c-5371-11e7-822f-172e500748e8.png)

The first argument of `Foo.bar` is dropped. But `Foo.baz` is not.

This code always drops a first argument of methods. This might be a bug fix.